biz.ganttproject.core/src/main/java/biz/ganttproject/core/chart/text/TimeFormatters.java

```diff
@@ -88,7 +88,38 @@ public static interface LocaleApi {
     DateFormat createDateFormat(String pattern);
     Locale getLocale();
     String i18n(String key);
-    Integer getWeekNumber(Date date);
+    ObservableProperty<Function<Date, Integer>> getWeekNumbering();
+  }
+
+  public static class ObservableProperty<T> {
+    private String myName;
+    private T myValue;
+    private List<PropertyChangeListener> myListeners = new ArrayList();;
+
+    public ObservableProperty (String name, T initValue) {
+      myName = name;
+      myValue = initValue;
+    }
+
+    public T getValue() {
+      return myValue;
+    }
+
+    public void setValue(T value) {
+      T oldValue = myValue;
+      myValue = value;
+      firePropertyChanged(oldValue, value);
+    }
+
+    public void addListener(PropertyChangeListener listener) {
+      myListeners.add(listener);
+    }
+
+    private void firePropertyChanged(T oldValue, T newValue) {
+      for (PropertyChangeListener listener : myListeners) {
```

Comment:
Perhaps it makes sense to use our own listener interface and event class as well. E.g., we can add type arguments to the event class and remove casting. Also, let's move this class into `biz.ganttproject.core.option` and rewrite in Kotlin ;) 

Oh, we have `ChangeValueListener` and `ChangeValueEvent` in `biz.ganttproject.core.option`, albeit without generic types.